src/System.Net.Mime/src/System/Net/Mime/Base64Stream.cs

```diff
@@ -0,0 +1,586 @@
+             15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25,255,255,255,255,255, // 5
+            255, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, // 6
+             41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51,255,255,255,255,255, // 7
+            255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // 8
+            255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // 9
+            255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // A
+            255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // B
+            255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // C
+            255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // D
+            255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // E
+            255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // F
+        };
+
+        private static readonly byte[] s_base64EncodeMap = new byte[] {
+             65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80,
+             81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 97, 98, 99,100,101,102,
+            103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,
+            119,120,121,122, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 43, 47,
+            61
+        };
+
+        private readonly int _lineLength;
+        private readonly Base64WriteStateInfo _writeState;
+        private ReadStateInfo _readState;
+
+        //the number of bytes needed to encode three bytes (see algorithm description in Encode method below)
+        private const int SizeOfBase64EncodedChar = 4;
+
+        //bytes with this value in the decode map are invalid
+        private const byte InvalidBase64Value = 255;
+
+        internal Base64Stream(Stream stream, Base64WriteStateInfo writeStateInfo) : base(stream)
+        {
+            _writeState = new Base64WriteStateInfo();
+            _lineLength = writeStateInfo.MaxLineLength;
+        }
+
+        internal Base64Stream(Stream stream, int lineLength) : base(stream)
+        {
+            _lineLength = lineLength;
+            _writeState = new Base64WriteStateInfo();
+        }
+
+        internal Base64Stream(Base64WriteStateInfo writeStateInfo) : base(new MemoryStream()) // TODO: added this... what should this be?
+        {
+            _lineLength = writeStateInfo.MaxLineLength;
+            _writeState = writeStateInfo;
+        }
+
+        private ReadStateInfo ReadState => _readState ?? (_readState = new ReadStateInfo());
+
+        internal Base64WriteStateInfo WriteState
+        {
+            get
+            {
+                Debug.Assert(_writeState != null, "_writeState was null");
+                return _writeState;
+            }
+        }
+
+        public override IAsyncResult BeginRead(byte[] buffer, int offset, int count, AsyncCallback callback, object state)
```

Comment:
The base Stream.ReadAsync/WriteAsync will end up calling these.  I don't know how this stream will be consumed and whether the call sites are/will use the XxAsync methods, but I expect if anything they're using the Begin/End ones directly.  Since this code is part of the 90% that's unused in this PR, I'm not going to add additional overrides.